### PR TITLE
gh-58032: Deprecate the argparse.FileType type converter

### LINKED ISSUE #58032: argparse.FileType opens a file and never closes it
BPO | [13824](https://bugs.python.org/issue13824)
--- | :---
Nosy | @merwok, @mitar, @matrixise, @MojoVampire, @remilapeyre, @Septatrix
Files | <li>[patch_3.diff](https://bugs.python.org/file31852/patch_3.diff "Uploaded as text/plain at 2013-09-23.20:26:03 by paul.j3")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2012-01-19.12:25:39.321>
labels = ['type-bug', 'library']
title = 'argparse.FileType opens a file and never closes it'
updated_at = <Date 2021-07-25.09:27:11.812>
user = 'https://bugs.python.org/DavidLayton'
```

bugs.python.org fields:
```python
activity = <Date 2021-07-25.09:27:11.812>
actor = 'Nils Kattenbeck'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2012-01-19.12:25:39.321>
creator = 'David.Layton'
dependencies = []
files = ['31852']
hgrepos = []
issue_num = 13824
keywords = ['patch']
message_count = 15.0
messages = ['151615', '152518', '152519', '152530', '166068', '198336', '198376', '215298', '341670', '341761', '341764', '341852', '342153', '342503', '398184']
nosy_count = 13.0
nosy_names = ['bethard', 'eric.araujo', 'mitar', 'Paolo.Elvati', 'manveru', 'Stefan.Pfeiffer', 'paul.j3', 'David.Layton', 'matrixise', 'josh.r', 'remi.lapeyre', 'sebix', 'Nils Kattenbeck']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue13824'
versions = ['Python 2.7']
```

</p></details>

### Linked PRs
* gh-113649
* gh-124664

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
* Issue: gh-58032

----
📚 Documentation preview 📚: https://cpython-previews--124664.org.readthedocs.build/

### CODE REVIEW

File: Lib/argparse.py [hunk 1]
```diff
@@ -39,7 +40,7 @@
 
     - FileType -- A factory for defining types of files to be created. As the
         example above shows, instances of FileType are typically passed as
-        the type= argument of add_argument() calls.
+        the type= argument of add_argument() calls. Deprecated.
```
Comment (Mariatta) [inline Lib/argparse.py]:
```suggestion
        the type= argument of add_argument() calls. Deprecated since Python 3.14
```

**Comment by Mariatta** [comment]
Should we add this to the `pending-removal-in-future.rst`?

File: Lib/test/test_argparse.py [hunk 2]
```diff
@@ -1744,27 +1744,44 @@ def convert_arg_line_to_args(self, arg_line):
...
+    with warnings.catch_warnings():
+        warnings.filterwarnings('ignore', 'FileType is deprecated',
+                                PendingDeprecationWarning, __name__)
+        return argparse.FileType(*args, **kwargs)
+
+
+class TestFileTypeDeprecation(TestCase):
+
+    def test(self):
+        parser = argparse.ArgumentParser()
```
Comment (savannahostrowski) [inline Lib/test/test_argparse.py]: Is this line required? It seems it isn't being used, unless there's some global state I'm forgetting about.

Comment (serhiy-storchaka) [reply Lib/test/test_argparse.py]: Good catch. It is absolutely not needed.
```suggestion
```

**Comment by serhiy-storchaka** [comment]
@Mariatta, could you please look again at this PR?

**Comment by hugovk** [comment]
Please see https://github.com/python/cpython/pull/125871 as a small docs update.

**Comment by StanFromIreland** [comment]
I'm a little late to the party, but I don't understand why `PendingDeprecationWarning` was used here. The docs, and warning message contradict the exception saying it *is* deprecated? Can it be updated to `DeprecationWarning` if it is deprecated, otherwise I think we should remove the `.. deprecated::` and rephrase the warning messages.

**Comment by serhiy-storchaka** [comment]
We use the `deprecated` directive even if there is no warning at runtime (they can be added latter). This differs from soft-deprecated where we do not plan runtime warning or deletion.

**Comment by StanFromIreland** [comment]
Yes, but I still don't understand why we're using PendingDeprecationWarning instead of DeprecationWarning here if it *is* deprecated?

**Pull request merged**